source/extensions/http/cache/file_system_http_cache/file_system_http_cache.h

```diff
@@ -21,13 +21,17 @@ namespace FileSystemHttpCache {
 using ConfigProto =
     envoy::extensions::http::cache::file_system_http_cache::v3::FileSystemHttpCacheConfig;
 
+class CacheEvictionThread;
+
```

Comment:
can we have a reference to the doc you wrote in the FileSystemCache class-comment here? Also is that doc up-to-date? If not, can we make it so?

Reply:
I'll make a `README.md` in this folder for it (or maybe a different-named markdown file so github doesn't shove a big doc in your face when you don't want it?)

Reply:
I think a link to the doc is OK; would that be easy?

Reply:
Added DESIGN.md, and I simplified the contents vs. the doc to exclude some of the speculative things that at this point feel unlikely, and exclude goals that require only changes to the cache filter not the cache itself.
([the doc](https://docs.google.com/document/d/1DWy1TE8RfV_0_fWs28DARS-GblUNuSRBqFxagOi6QPQ/edit#heading=h.7sus632d57si) in case you want to compare.)

Reply:
sure, sg. Can you add a ref to where that will be here?

Reply:
Added reference to DESIGN.md on the class description for both FileSystemHttpCache and CacheEvictionThread.